public/app/features/explore/LogsVolumePanel.tsx

```diff
@@ -29,56 +30,20 @@ type Props = {
   eventBus: EventBus;
 };
 
-function createVisualisationData(
-  logLinesBased: DataQueryResponse | undefined,
-  logLinesBasedVisibleRange: AbsoluteTimeRange | undefined,
-  fullRangeData: DataQueryResponse | undefined,
-  absoluteRange: AbsoluteTimeRange
-):
-  | {
-      logsVolumeData: DataQueryResponse;
-      fullRangeData: boolean;
-      range: AbsoluteTimeRange;
-    }
-  | undefined {
-  if (fullRangeData !== undefined) {
-    return {
-      logsVolumeData: fullRangeData,
-      fullRangeData: true,
-      range: absoluteRange,
-    };
-  }
-
-  if (logLinesBased !== undefined) {
-    return {
-      logsVolumeData: logLinesBased,
-      fullRangeData: false,
-      range: logLinesBasedVisibleRange || absoluteRange,
-    };
-  }
-
-  return undefined;
-}
-
 export function LogsVolumePanel(props: Props) {
   const { width, timeZone, splitOpen, onUpdateTimeRange, onLoadLogsVolume, onHiddenSeriesChanged } = props;
   const theme = useTheme2();
   const styles = useStyles2(getStyles);
   const spacing = parseInt(theme.spacing(2).slice(0, -2), 10);
   const height = 150;
 
-  const data = createVisualisationData(
-    props.logLinesBasedData,
-    props.logLinesBasedDataVisibleRange,
-    props.logsVolumeData,
-    props.absoluteRange
-  );
-
-  if (data === undefined) {
+  if (props.logsVolumeData === undefined) {
     return null;
   }
 
-  const { logsVolumeData, fullRangeData, range } = data;
+  const logsVolumeData = props.logsVolumeData;
+  const range = logsVolumeData.data[0]?.meta?.custom?.absoluteRange || props.absoluteRange;
+  const fullRangeData = logsVolumeData.data[0]?.meta?.custom?.logsVolumeType !== LogsVolumeType.Limited;
```

Comment:
I would either rename `fullRangeData` to `isFullRangeData` as the value is boolean. Or (in my opinion better), do `const logsVolumeType = logsVolumeData.data[0]?.meta?.custom?.logsVolumeType` and use that below.

Reply:
Good point! Fixed in 7ce66774d2af6dc6444f37fe358261a7b321592a